src/components/SuggestionTags/Tags/PriceImpactWarningSuggestion.tsx

```diff
@@ -0,0 +1,200 @@
+
+import BasePill, { DANGER_CLASS, WARNING_CLASS } from './BasePill';
+import InfoIconSVG from 'src/icons/InfoIconSVG';
+
+import Link from 'next/link';
+
+import RightArrowIcon from 'src/icons/RightArrowIcon';
+
+import { QuoteResponse } from '@jup-ag/react-hook';
+import { useUSDValue } from 'src/contexts/USDValueProvider';
+import { useSwapRouteInfo } from '../hooks/useSwapInfo';
+import { useMobile } from 'src/hooks/useMobile';
+import JupButton from 'src/components/JupButton';
+import { cn } from 'src/misc/cn';
+import PopoverTooltip from 'src/components/Tooltip/PopoverTooltip';
+
+interface Props {
+  isWarning: boolean;
+  isDanger: boolean;
+  birdeyeRate: number | null;
+  isHighPriceImpact: boolean;
+  priceDifferencePct: number;
+  fromTokenInfo: TokenInfo;
+  toTokenInfo: TokenInfo;
+  quoteResponse: QuoteResponse;
+}
+
+const PriceImpactWarningSuggestion = ({
+  birdeyeRate,
+  isWarning,
+  isDanger,
+  isHighPriceImpact,
+  priceDifferencePct,
+  fromTokenInfo,
+  toTokenInfo,
+  quoteResponse,
+}: Props) => {
+  const { tokenPriceMap } = useUSDValue();
+  const { inAmount, outAmount, priceImpactPct } = quoteResponse;
+
+  const tradeValue = useMemo(
+    () => new Decimal(inAmount.toString()).div(10 ** fromTokenInfo.decimals),
+    [inAmount, fromTokenInfo.decimals],
+  );
+
+  const fromTokenValue = useMemo(
+    () => tradeValue.mul(tokenPriceMap[fromTokenInfo.address || '']?.usd || 0),
+    [tradeValue, fromTokenInfo.address, tokenPriceMap],
+  );
+
+  const toTokenValue = useMemo(
+    () =>
+      new Decimal(outAmount.toString())
+        .div(10 ** toTokenInfo.decimals)
+        .mul(tokenPriceMap[toTokenInfo.address || '']?.usd || 0),
+    [outAmount, toTokenInfo.address, toTokenInfo.decimals, tokenPriceMap],
+  );
+
+  const shouldPromptDca = useMemo(() => fromTokenValue.gte(1_000), [fromTokenValue]);
+
+  const [reverse, setReverse] = React.useState(true);
+
+  const onReverse: React.MouseEventHandler = React.useCallback((event) => {
+    event.stopPropagation();
+    setReverse((prevState) => !prevState);
+  }, []);
+
+  const swapRouteInfo = useSwapRouteInfo();
+  const renderWarningColor = useMemo(
+    () => isWarning || swapRouteInfo?.isMediumImpact,
+    [isWarning, swapRouteInfo?.isMediumImpact],
+  );
+
+  const renderDangerColor = useMemo(
+    () => isDanger || swapRouteInfo?.isHighImpact,
+    [isDanger, swapRouteInfo?.isHighImpact],
+  );
+
+  const isMobile = useMobile();
+
+  const { pillTitle, title, description } = useMemo(() => {
+    if (isHighPriceImpact) {
+      return {
+        pillTitle: `${new Decimal(priceImpactPct).mul(100).toNumber().toFixed(2)}% Price Impac`,
+        title: shouldPromptDca ? `Looks like your trade size is too large.` : `Looks like you are getting a bad rate.`,
+        description: shouldPromptDca ? (
+          <Link
+            shallow
+            href={`/dca/${fromTokenInfo.address}-${toTokenInfo.address}?${new URLSearchParams({
```

Comment:
this one is wrong, should be https://jup.ag, right now it just reloads t he widget